File: src/test/java/it/marketplace/microservices/controller/paymentOrder/GetPaymentOrderControllerTest.java
package it.marketplace.microservices.controller.paymentOrder;

import it.marketplace.microservices.common.dto.PaymentOrderDto;
import it.marketplace.microservices.common.resource.PaymentOrderResource;
import it.marketplace.microservices.config.exception.ServiceException;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;
import org.springframework.http.ResponseEntity;



import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.mockito.Mockito.when;
import static org.mockito.Mockito.verify;

class GetPaymentOrderControllerTest {

    @Mock
    private PaymentOrderService service;

    @InjectMocks
    private GetPaymentOrderController controller;

    @BeforeEach
    void setUp() {
        MockitoAnnotations.openMocks(this);
    }

    @Test
    void shouldReturnPaymentOrders_WhenFindByUserEmail_ThenArrangeActAssert() throws ServiceException {
        // Arrange
        String email = "test@email.com";
        PaymentOrderDto dto = new PaymentOrderDto();
        List<PaymentOrderDto> dtos = List.of(dto);
        when(service.findOrderByEmail(email)).thenReturn(dtos);
        List<PaymentOrderResource> expected = dtos.stream().map(PaymentOrderMapper::toResource).toList();

        // Act
        ResponseEntity<List<PaymentOrderResource>> response = controller.find(email);

        // Assert
        assertEquals(200, response.getStatusCodeValue());
        assertEquals(expected, response.getBody());
        verify(service).findOrderByEmail(email);
    }

    @Test
    void shouldReturnAllPaymentOrders_WhenFindAll_ThenArrangeActAssert() throws ServiceException {
        // Arrange
        PaymentOrderDto dto = new PaymentOrderDto();
        List<PaymentOrderDto> dtos = List.of(dto);
        when(service.findAll()).thenReturn(dtos);
        List<PaymentOrderResource> expected = dtos.stream().map(PaymentOrderMapper::toResource).toList();

        // Act
        ResponseEntity<List<PaymentOrderResource>> response = controller.findAll();

        // Assert
        assertEquals(200, response.getStatusCodeValue());
        assertEquals(expected, response.getBody());
        verify(service).findAll();
    }
}

File: src/test/java/it/marketplace/microservices/controller/paymentOrder/GetPaymentOrderControllerTest.kt
package it.marketplace.microservices.controller.paymentOrder

import it.marketplace.microservices.common.dto.toResource
import it.marketplace.microservices.common.resource.PaymentOrderResource
import it.marketplace.microservices.config.exception.ServiceException
import it.marketplace.microservices.service.PaymentOrderService
import it.marketplace.microservices.utils.BaseTest
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import org.mockito.InjectMocks
import org.mockito.Mock
import org.mockito.Mockito
import org.mockito.Mockito.mock
import org.mockito.MockitoAnnotations
import org.springframework.http.ResponseEntity
import kotlin.test.assertEquals

class GetPaymentOrderControllerTest : BaseTest() {
    @Mock
    private val service: PaymentOrderService = mock()

    @InjectMocks
    private val controller: GetPaymentOrderController = mock()

    @BeforeEach
    fun setUp() {
        MockitoAnnotations.openMocks(this)
    }

    @Test
    @Throws(ServiceException::class)
    fun shouldReturnPaymentOrders_WhenFindByUserEmail_ThenArrangeActAssert() {
        // Arrange
        val email = "test@email.com"
        val dto = mockPaymentOrderDto()
        val dtos = listOf(dto)
        val expected = listOf(dto.toResource())
        Mockito.`when`(service.findOrderByEmail(email)).thenReturn(dtos)

        // Act
        val response: ResponseEntity<kotlin.collections.List<PaymentOrderResource?>?> = controller.find(email)

        // Assert
        assertEquals(200, response.statusCode.value())
        assertEquals(expected, response.getBody())
        Mockito.verify(service).findOrderByEmail(email)
    }

    @Test
    @Throws(ServiceException::class)
    fun shouldReturnAllPaymentOrders_WhenFindAll_ThenArrangeActAssert() {
        // Arrange
        val dto = mockPaymentOrderDto()
        val dtos = listOf(dto)
        val expected = listOf(dto.toResource())
        Mockito.`when`(service.findAll()).thenReturn(dtos)

        // Act
        val response: ResponseEntity<List<PaymentOrderResource?>?> = controller.findAll()

        // Assert
        assertEquals(200, response.statusCode.value())
        assertEquals(expected, response.getBody())
        Mockito.verify(service).findAll()
    }
}